### DISCUSSION TITLE
Classifying local channels from the US playlist.

### DISCUSSION THREAD

**Discussion by BellezaEmporium** [iptv-org/community#289] [Polls]
I am currently sorting out most of the channels to make sure the playlists are easier to read.

Though I face a little issue that'll need an community answer.

**Comment by benevenstanciano** [comment]
I think there isn't a single great solution, since the problem has many layers. You've got local channels that stream their full service, others that only stream news, others have a specific streaming version that loops local shows; then you've got the issue that although they're part of a network they're affiliates and stream only local programming so ultimately the network doesn't matter, though it helps people identify the service, etc... It's a real mess to try to fix. I did vote for having a separate local playlist, but I think they're also OK to stay where they are.

**Reply by BellezaEmporium** [reply]
Well, I envision things like this : 

- Local channels are often viewed as "the stuff they put off antennas and are different on every state". They are, however, linked to some group unless they're independent (myNetworkTV channels as an example).
- I sincerely think they should be put under their specific groups however since they're airing different programmation than the main sources (East and West feeds are different, more local programmation is brought out), they should have their own playlist.

However I do agree this is a big schmiblick to handle, due to the fact that it's "a group channel, but not their programmation".

**Comment by AntiPontifex** [comment]
Another issue is stations that have multiple streams. It would be nice if somehow the playlist could be arranged/programmed so it is only listed _once_ with the user then given the option of choosing the wanted stream (ordered by stream bitrate?).

**Reply by BellezaEmporium** [reply]
The choice of stations being put first is more a matter of quality. After all, a better quality stream will, most of the time, be linked to a bigger stream bitrate, due to the number of informations being bigger to push.

**Reply by AntiPontifex** [reply]
Oh, I agree; but users with bandwidth issues would be able to make decisions based on need. Just a suggestion, Lane ;)

**Comment by Sphinxroot** [comment]
good idea or not


us.m3u  ( international / national )
us.local.m3u (local / affiliated ) 
us.peg.m3u ( public access television / public education television / Gov / Municipal )

state / Cities name / call sign /  channel name ... cbs fox abc nbc ... / channel number

Exemple of classing them on naming
state:
#EXTINF:-1,Arkansas | Fayetteville | FPTV
or Short
#EXTINF:-1,AR | Fayetteville | FPTV
https://www.youtube.com/channel/UCT3Dk_ZPcoR7CeHM_Wwb3QA/live

having the state name or code at first might help to sorted them for quick search and access

**Reply by BellezaEmporium** [reply]
I could make it so it matches the way EPG is sorted out.

**Comment by AntiPontifex** [comment]
That's an idea I can get behind - but it is still a tremendous amount of work. Excellent suggestion, though, sphinx.

**Comment by loucasal** [comment]
Completely in favour - and this is relevant for more than just the US.
To be clear, I think this should be made available to users, not just used for structuring the repo (similarly to my suggestion in iptv-org/iptv#6164).

**Reply by BellezaEmporium** [reply]
That would require some little script rework. I could rewrite the scripts though I don't have the time to do so. Maybe @freearhey , if he's OK with the idea, could take a look at it and fiddle with the playlists.

**Comment by AntiPontifex** [comment]
Sounds good: looks like you have found yourself a solution, Lane. If you need any help with this, you know how to contact me.

**Comment by BellezaEmporium** [comment]
The option that's been kept is the "create a us_local.m3u".

As correctly mentioned by @benevenstanciano , there are some channels that strictly show news programs. Either I create a "us_localnews.m3u" and "us_localota.m3u", either i put all of them onto the us_local.m3u, and we make sure the name of the channel is explicit  ((OTA) and (News) will be added next to the name of the channel).

**Comment by AntiPontifex** [comment]
so, that's where the peg channels are going to be, too, or are they going to be in separate .m3u?

**Reply by BellezaEmporium** [reply]
Public, Educational and Governmental channels, if local, will be on us_local.m3u since it's mainly information linked to the area.

**Comment by AntiPontifex** [comment]
Gotcha. Thanks, Lane.

**Comment by BellezaEmporium** [comment]
> I am currently sorting out most of the channels to make sure the playlists are easier to read.
> 
> Though I face a little issue that'll need an community answer.

Vote has been taken onto account, file here : https://github.com/iptv-org/iptv/blob/caf02c753b60f277aee405d666d852d4cbe1e234/streams/us_local.m3u
